### PR TITLE
added target sum triplets #585

### PR DESCRIPTION
**Pull request by NEEHITGOYAL**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?
YES
### Description
added solution to find all sets of 3 number in an array whose sum is equal to the target given by user
Time Complexity: O(n2)
space: O(1)
### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

fixed issue #585

### CODE REVIEW

**Comment by NEEHITGOYAL** [comment]
kindly review and merge it @HarshCasper @iamrajiv @SKAUL05 @Kajol-Kumari @plazzy99 @siddharth25pandey

File: C-Plus-Plus/cp/target sum triplets.cpp [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+            // if current sum is greater than  'sum', then decrement the right index 
+            else
+                right--; 
+        } 
+	}
+}
+
+//Main function
+int main() 
+{
```
Comment (vatsalkesarwani12) [inline C-Plus-Plus/cp/target sum triplets.cpp]: add proper messages for input and output

**Review by vatsalkesarwani12 — CHANGES_REQUESTED** [review]
edit readme

File: C-Plus-Plus/cp/target sum triplets.cpp (hunk 1 above)
Comment (NEEHITGOYAL) [reply C-Plus-Plus/cp/target sum triplets.cpp]: added messages as well as updated readme kindly review and merge it

**Review by vatsalkesarwani12 — APPROVED** [review]
LGTH 👍

File: C-Plus-Plus/cp/target_sum_triplets.cpp [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+                    right--; 
+                } 
+	 }
+}
+
+//Main function
+int main() 
+{
+	int end,input[1000],target;
+	cout<<"Enter the no. of elements in array you want to enter"<<endl;
```
Comment (Abhijit2505) [inline C-Plus-Plus/cp/target_sum_triplets.cpp:43]: Please use dynamic arrays because your code is only for an array of size 1000

**Review by Abhijit2505 — CHANGES_REQUESTED** [review]

File: C-Plus-Plus/cp/target_sum_triplets.cpp (hunk 2 above)
Comment (NEEHITGOYAL) [reply C-Plus-Plus/cp/target_sum_triplets.cpp:43]: done @Abhijit2505  sir

**Review by Abhijit2505 — APPROVED** [review]
LGTM

**Comment by iamrajiv** [comment]
@all-contributors please add @NEEHITGOYAL for code

**Pull request merged**